src/Compilers/CSharp/Test/Semantic/Semantics/NullableContextTests.cs

```diff
@@ -413,8 +430,1174 @@ void verify(CSharpParseOptions parseOptions, bool expectedFlowState, params stri
+                comp.NullableAnalysisData = new ConcurrentDictionary<object, NullableWalker.Data>();
+
+                if (isNullableEnabledForMethod)
+                {
+                    comp.VerifyDiagnostics(
+                        // (4,30): warning CS8625: Cannot convert null literal to non-nullable reference type.
+                        //     static void Main() { A.M(null); }
+                        Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null"));
+                }
+                else
+                {
+                    comp.VerifyDiagnostics();
+                }
+
+                var actualAnalyzedKeys = GetNullableDataKeysAsStrings(comp.NullableAnalysisData, requiredAnalysis: true);
+                Assert.Equal(actualAnalyzedKeys.Contains("Main"), isNullableEnabledForMethod);
+
+                var tree = (CSharpSyntaxTree)comp.SyntaxTrees[0];
+                var syntaxNodes = tree.GetRoot().DescendantNodes();
+                verifyContextState(tree, syntaxNodes.OfType<ClassDeclarationSyntax>().Single(), classDirectives.ExpectedWarningsState, classDirectives.ExpectedAnnotationsState);
+                verifyContextState(tree, syntaxNodes.OfType<MethodDeclarationSyntax>().Single(), expectedWarningsStateForMethod, expectedAnnotationsStateForMethod);
+
+                static NullableContextState.State combineState(NullableContextState.State currentState, NullableContextState.State previousState)
+                {
+                    return currentState == NullableContextState.State.Unknown ? previousState : currentState;
+                }
+
+                static void verifyContextState(CSharpSyntaxTree tree, CSharpSyntaxNode syntax, NullableContextState.State expectedWarningsState, NullableContextState.State expectedAnnotationsState)
+                {
+                    var actualState = tree.GetNullableContextState(syntax.SpanStart);
+                    Assert.Equal(expectedWarningsState, actualState.WarningsState);
+                    Assert.Equal(expectedAnnotationsState, actualState.AnnotationsState);
+                }
+
+                static bool isNullableEnabled(NullableContextState.State state, bool isNullableEnabledForProject)
+                {
+                    return state switch
+                    {
+                        NullableContextState.State.Enabled => true,
+                        NullableContextState.State.Disabled => false,
+                        _ => isNullableEnabledForProject,
+                    };
+                }
+            }
+        }
+
+        [Fact]
+        [WorkItem(49746, "https://github.com/dotnet/roslyn/issues/49746")]
+        public void AnalyzeMethodsInEnabledContextOnly_02()
+        {
+            var source =
+@"#pragma warning disable 219
+static class Program
+{
+#nullable disable
+    static void M1
+#nullable enable
+        ()
+    {
+        object obj = null;
```

Comment:
Consider adding a `_ = obj` line and verifying the results of `GetTypeInfo` on the `obj` reference for these.

Reply:
Including nullability explicitly.

---
In reply to: [546890154](https://github.com/dotnet/roslyn/pull/49953#discussion_r546890154) [](ancestors = 546890154)

Reply:
`GetTypeInfo()` is included in `AnalyzeMethodsInEnabledContextOnly_MethodBodySemanticModel_01`, etc.

---
In reply to: [546890244](https://github.com/dotnet/roslyn/pull/49953#discussion_r546890244) [](ancestors = 546890244,546890154)